src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -1477,31 +1461,58 @@ public override void WriteJson(JsonWriter writer, Stream? value, JsonSerializer
         }
 
         [DebuggerDisplay("{" + nameof(DebuggerDisplay) + "}")]
-        private class RpcMarshalableImplicitConverter : JsonConverter
+        private class RpcMarshalableReadConverter : JsonConverter
```

Comment:
@AArnott, I split `RpcMarshalableImplicitConverter` into two.
`RpcMarshalableReadConverter` resolves the type options during deserialization, but cannot write. This is added to the formatter's converter list.
`RpcMarshalableConverter` has properties for the interface type and the corresponding options (like the old `RpcMarshalableImplicitConverter` did). This is necessary because `WriteJson` doesn't take the definition type and trying to find the type at runtime would misbehave in case an object implements multiple marshalable interfaces (this is what was breaking the tests on Mono).